### PR TITLE
Refactor `module`

### PR DESCRIPTION
**Pull request by ordabayevy**
- Combine `PyroTrainingPlan` and `DummyTrainingPlan`
- Import optimizer and scheduler if they are passed as strings
- Add `BaseModule` and `BasePyroModule` base classes
- Rename `ProbabilisticPCAPyroModule` to `ProbabilisticPCA`
- Add `forward` method to the `ProbabilisticPCA`
- Use `register_buffer` for `torch.nn.Module` attributes so that they are saved at checkpointing
- Fix type hints for `mean_g`, `optim_fn`, `scheduler_fn`
- Add shape hints for `OnePassMeanVarStd` attributes (e.g. `mean` -> `mean_g`)

### CODE REVIEW

File: scvid/train/training_plan.py [hunk 1]
```diff
@@ -28,43 +25,57 @@ class PyroTrainingPlan(pl.LightningModule):
...
+            class_module, class_name = scheduler_fn.rsplit(".", 1)
+            scheduler_module = import_module(class_module)
+            self.scheduler_fn = getattr(scheduler_module, class_name)
+        else:
+            self.scheduler_fn = scheduler_fn
 
-        self.optim_fn = torch.optim.Adam if optim_fn is None else optim_fn
         optim_kwargs = {} if optim_kwargs is None else optim_kwargs
         if "lr" not in optim_kwargs:
             optim_kwargs["lr"] = 1e-3
```
Comment (mbabadi) [inline scvid/train/training_plan.py:56]: Magic number in the middle of a method-- move it to the initializer kwargs with a default or a _consts.py.

**Review by mbabadi — CHANGES_REQUESTED** [review]
Nicely done!

**Comment by ordabayevy** [comment]
@mbabadi thanks! Added `default_lr` kwarg and fixed some mypy errors.

**Review by mbabadi — APPROVED** [review]
